pkg/config/constants.go

```diff
@@ -172,24 +172,86 @@ const DefaultPrometheusConfiguration = `
 lowercaseOutputName: true
 attrNameSnakeCase: true
 rules:
-  - pattern: metrics<name=(\S+)/(\S+)\.driver\.(BlockManager|DAGScheduler)\.(\S+)><>Value
+  - pattern: metrics<name=(\S+)/(\S+)\.driver\.(BlockManager|DAGScheduler|jvm)\.(\S+)><>Value
     name: spark_driver_$3_$4
+    type: GAUGE
     labels:
       app_namespace: "$1"
       app_name: "$2"
   - pattern: metrics<name=(\S+)/(\S+)\.driver\.(\S+)\.StreamingMetrics\.streaming\.(\S+)><>Value
     name: spark_streaming_driver_$4
+    type: GAUGE
     labels:
       app_namespace: "$1"
       app_name: "$2"
   - pattern: metrics<name=(\S+)/(\S+)\.driver\.spark\.streaming\.(\S+)\.(\S+)><>Value
     name: spark_structured_streaming_driver_$4
+    type: GAUGE
     labels:
       app_namespace: "$1"
       app_name: "$2"
       query_name: "$3"
   - pattern: metrics<name=(\S+)/(\S+)\.(\S+)\.executor\.(\S+)><>Value
     name: spark_executor_$4
+    type: GAUGE
+    labels:
+      app_namespace: "$1"
+      app_name: "$2"
+      executor_id: "$3"
+  - pattern: metrics<name=(\S+)/(\S+)\.driver\.DAGScheduler\.(.*)><>Count
+    name: spark_driver_DAGScheduler_$3_count
+    type: COUNTER
+    labels:
+      app_namespace: "$1"
+      app_name: "$2"
+  - pattern: metrics<name=(\S+)/(\S+)\.driver\.HiveExternalCatalog\.(.*)><>Count
+    name: spark_driver_HiveExternalCatalog_$3_total
+    type: COUNTER
+    labels:
+      app_namespace: "$1"
+      app_name: "$2"
+  - pattern: metrics<name=(\S+)/(\S+)\.driver\.CodeGenerator\.(.*)><>Count
+    name: spark_driver_CodeGenerator_$3_count
+    type: COUNTER
+    labels:
+      app_namespace: "$1"
+      app_name: "$2"
+  - pattern: metrics<name=(\S+)/(\S+)\.driver\.LiveListenerBus\.(.*)><>Count
+    name: spark_driver_LiveListenerBus_$3_count
+    type: COUNTER
+    labels:
+      app_namespace: "$1"
+      app_name: "$2"
+  - pattern: metrics<name=(\S+)/(\S+)\.driver\.LiveListenerBus\.(.*)><>Value
+    name: spark_driver_LiveListenerBus_$3
+    type: GAUGE
+    labels:
+      app_namespace: "$1"
+      app_name: "$2"
+  - pattern: metrics<name=(\S+)/(\S+)\.(.*)\.executor\.(.*)><>Count
+    name: spark_executor_$4_total
```

Comment:
Looks like some metrics use `count`, and this and some others use `total`. Should we make them consistent?

Reply:
Sounds good. Fixed.